docs/csharp/language-reference/builtin-types/struct.md

```diff
@@ -77,14 +77,40 @@ When you design a structure type, you have the same capabilities as with a [clas
 
 - You can't declare a parameterless constructor. Every structure type already provides an implicit parameterless constructor that produces the [default value](default-values.md) of the type.
 
+  > [!NOTE]
+  > Beginning with C# 10.0, you can declare a parameterless constructor in a structure type. For more information, see the [Parameterless constructors and field initializers](#parameterless-constructors-and-field-initializers) section.
+
 - You can't initialize an instance field or property at its declaration. However, you can initialize a [static](../keywords/static.md) or [const](../keywords/const.md) field or a static property at its declaration.
 
+  > [!NOTE]
+  > Beginning with C# 10.0, you can initialize an instance field or property at its declaration. For more information, see the [Parameterless constructors and field initializers](#parameterless-constructors-and-field-initializers) section.
+
 - A constructor of a structure type must initialize all instance fields of the type.
 
 - A structure type can't inherit from other class or structure type and it can't be the base of a class. However, a structure type can implement [interfaces](../keywords/interface.md).
 
 - You can't declare a [finalizer](../../programming-guide/classes-and-structs/destructors.md) within a structure type.
 
+## Parameterless constructors and field initializers
+
+Beginning with C# 10.0, you can declare a parameterless instance constructor in a structure type, as the following example shows:
+
+:::code language="csharp" source="snippets/shared/StructType.cs" id="ParameterlessConstructor":::
+
+As the preceding example shows, the [default value expression](../operators/default.md) ignores a parameterless constructor and produces the default value of a structure type, which is the value produced by setting all value-type fields to their [default values](default-values.md) and all reference-type fields to `null`. Structure-type array instantiation also ignores a parameterless constructor and produces an array populated with the default values of a structure type.
+
+Beginning with C# 10.0, you can also initialize an instance field or property at its declaration, as the following example shows:
+
+:::code language="csharp" source="snippets/shared/StructType.cs" id="FieldInitializer":::
+
+As the preceding example shows, if you don't declare a parameterless constructor, the implicit parameterless constructor produces the default value of a structure type, regardless of field initializers. However, if a structure type has no explicit instance constructors and has field initializers, the compiler synthesizes a public parameterless constructor that performs the specified field initializations, as the following example shows:
```

Comment:
The current wording took me a few times to understand correctly. How about this alternative:

```suggestion
The preceding example demonstrates that the implicit parameterless constructor performs the specified field initializations only if no instance constructors are declared. If you write an explicit parameterless constructor, that constructor must assign all instance fields in its body. 
```

Reply:
@BillWagner 

> The preceding example demonstrates that the implicit parameterless constructor performs the specified field initializations only if no instance constructors are declared.

As far as I understand, that's not correct, because there is a difference between the *implicit* parameterless constructor that *always* returns a zeroed instance and the *synthesized* parameterless constructor. This is how I've read this section in spec: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/parameterless-struct-constructors#constructors

Or you're fine to mix the implicit and synthesized parameterless constructors? Not sure, as the synthesized constructor is still explicit, but just written by the compiler.

> If you write an explicit parameterless constructor, that constructor must assign all instance fields in its body.

That's not correct, as the instance fields with field initializers are not required to be assigned in the body of the constructor. In general, I think it's fine to omit that sentence, as the corresponding compiler error message is clear enough.

I'll need to think more about wording here, I see.

Reply:
@BillWagner I've rewritten the paragraph. What do you think of the new version?

Reply:
Looking back at my sharplab.io history, I see what I did wrong. I had one field without an initializer when I was testing. Reviewing again.